huntsman/camera/pyro.py

```diff
@@ -177,73 +185,51 @@ def take_exposure(self,
-
-        # Clear event now to prevent any other exposures starting before this one is finished.
-        self._exposure_event.clear()
-
-        # Make sure proxy is in async mode
-        Pyro4.asyncproxy(self._proxy, asynchronous=True)
-
         # Start the exposure
-        self.logger.debug(f'Taking {seconds} second exposure on {self.name}: {filename}')
+        self.logger.debug(f'Taking {seconds} second exposure on {self}: {filename}')
 
         # Remote method call to start the exposure
-        exposure_result = self._proxy.take_exposure(seconds=seconds,
-                                                    filename=filename,
-                                                    dark=bool(dark),
-                                                    *args,
-                                                    **kwargs)
-
-        # Start a thread that will set an event once exposure has completed
-        exposure_thread = Timer(interval=get_quantity_value(seconds, u.s) + self.readout_time,
-                                function=self._async_wait,
-                                args=(exposure_result,
-                                      'exposure',
-                                      self._exposure_event,
-                                      self._timeout))
-        exposure_thread.start()
+        self._proxy.take_exposure(seconds=seconds,
+                                  filename=filename,
+                                  dark=bool(dark),
+                                  *args,
+                                  **kwargs)
 
+        max_wait = get_quantity_value(seconds, u.second) + self.readout_time + self._timeout
         if blocking:
-            self._exposure_event.wait()
+            success = self._exposure_event.wait(timeout=max_wait)
+            if not success:
+                self._timeout_response("exposure")
+        else:
+            # If the remote exposure fails after starting in such a way that the event doesn't
+            # doesn't get set then calling code could wait forever. Have a local timeout thread
+            # to be safe.
+            timeout_thread = Timer(interval=max_wait,
+                                   function=self._timeout_response,
+                                   args=("exposure",))
+            timeout_thread.start()
 
         return self._exposure_event
 
-    def autofocus(self,
-                  seconds=None,
-                  focus_range=None,
-                  focus_step=None,
-                  thumbnail_size=None,
-                  keep_files=None,
-                  take_dark=None,
-                  merit_function='vollath_F4',
-                  merit_function_kwargs={},
-                  mask_dilations=None,
-                  coarse=False,
-                  make_plots=False,
-                  blocking=False,
-                  timeout=None,
-                  *args, **kwargs):
+    def autofocus(self, blocking=False, *args, **kwargs):
```

Comment:
Is it still appropriate to have all the params documented here?

Reply:
I thought so, as while this is just a duplication of the docstring from the `Camera` base class (and the `Focuser` base class) it is a public method of a potentially user facing class so having the docstring here might help with code editor inline help/tooltips/autocomplete/etc.